src/cpp/cpp_declarator_converter.cpp

```diff
@@ -326,11 +326,10 @@ void cpp_declarator_convertert::handle_initializer(
 {
   exprt &value=declarator.value();
 
-  // moves member initializers into 'value'
-  cpp_typecheck.move_member_initializers(
-    declarator.member_initializers(),
-    symbol.type,
-    value);
+  // moves member initializers into 'value' - only methods have these
+  if(!declarator.member_initializers().get_sub().empty())
```

Comment:
You add an initializer... only if there is already one?

Reply:
The method called will actually only do anything when this condition evaluates to true. Not checking this here would invoke the method for non-code-type things, which makes the `to_code_type` fail, even though the invoked method would just be a no-op.

Reply:
So really not a change in behaviour, just a necessary condition to make the subsequent `to_code_type` (which is now necessary) pass.

Reply:
That suggests you should check whether it has id() == ID_code instead?

Reply:
Indeed. Done.